nio/client/async_client.py

```diff
@@ -3441,6 +3441,35 @@ async def room_upgrade(self,
 
         return RoomUpgradeResponse(new_room.room_id)
 
+    @logged_in
+    async def update_room_topic(
+        self,
+        room_id: str,
+        topic: str,
+        state_key: str = "",
+    ) -> Union[RoomPutStateResponse, RoomPutStateError]:
+        """Update the room topic
+
+        Returns either a `RoomPutStateResponse` if the request was successful
+        or a `RoomPutStateError` if there was an error with the request.
+
+        Args:
+            room_id (str): The room id of the room to be updated.
+            topic (str): The new room topic.
+            state_key (str): The key of the state event to send. Defaults to an empty string.
+        """
+
+        method, path, data = Api.update_room_topic(
+            self.access_token,
+            room_id,
+            topic,
+            state_key=state_key,
+        )
+
+        return await self._send(
```

Comment:
You should use the generic [`room_put_state`](https://matrix-nio.readthedocs.io/en/latest/nio.html#nio.AsyncClient.room_put_state) method for this.

Reply:
Fixed in e0fdf67.